src/weather_man.js

```diff
@@ -8,23 +8,35 @@ class WeatherMan extends React.Component {
     super(props)
     this.state = {
       city: "",
-      weather: "",
+      weatherForecast: "",
     }
+    this.registerCity = this.registerCity.bind(this);
+    this.updateWeatherForecast = this.updateWeatherForecast.bind(this);
   }
 
-  componentDidMount() {
-    getDataFor(this.props.city)
+  registerCity(event) {
+    this.setState({
+      city: event.target.value,
+    })
+  }
+
+  updateWeatherForecast() {
+    getDataFor(this.state.city)
       .then(data => {
 	this.setState({
-	  city: data.name,
-	  weather: data.weather[0].description,
-	});
-      });
+	  weatherForecast: "The weather in " + this.state.city + " now is: " + data.weather[0].description.toUpperCase(),
```

Comment:
I would probably use string interpolation with back ticks (they call it "template literals") instead of using `+`.  That makes it a bit more readable https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals